src/wrappers/themis/python/pythemis/scell.py

```diff
@@ -24,11 +24,42 @@
 themis = cdll.LoadLibrary(find_library('themis'))
 
 
+class SecureCellError(ThemisError):
+    def __init__(self, message, error_code=THEMIS_CODES.INVALID_PARAMETER):
+        message = 'Secure Cell: ' + message
+        super(SecureCellError, self).__init__(error_code, message)
+
+
 class SCellSeal(object):
+    def __new__(cls, key=None, passphrase=None):
+        """
+        Make a new Secure Cell in Seal mode.
+
+        You must specify either key= or passphrase= keyword argument.
+
+        :type key: bytes
+        :type passphrase: Union(str, bytes)
+        """
+        if key is None and passphrase is None:
+            raise SecureCellError('missing key or passphrase')
+        if key is not None and passphrase is not None:
+            raise SecureCellError('key and passphrase cannot be specified at the same time')
+        if key is not None:
+            return object.__new__(SCellSeal)
+        if passphrase is not None:
+            return object.__new__(SCellSealPassphrase)
+
     def __init__(self, key):
+        """
+        Make a new Secure Cell in Seal mode with a master key.
+
+        :type key: bytes
+        """
         if not key:
```

Comment:
> if we check it in new already, why it checked second time?

Better safe than sorry :) I left it there just in case there is a way to call `__init__` without calling `__new__`.